### ISSUE TITLE
Document verify_bucket_exists parameter for S3

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/documentation#9229]
We currently only have the autocreate parameter in the docs, but `verify_bucket_exists` should be documented as well.

**Comment by rhtot** [comment]
Some details on how we have encountered an issue related to this topic:

We have encountered a bucket creation failure error accidentally during a network issue which caused some requests to time out on connection attempts to the S3 backend.

Below is a sample of the exception from one of our log entries.

`"exception": {
    "CustomMessage": "Could not update object urn:oid:3999248800 for files/examplepath",
    "Exception": "Exception",
    "Message": "Creation of bucket \"examplebucket\" failed. Error executing \"CreateBucket\" on \"https://exampleprovider.com/examplebucket\"; AWS HTTP error: cURL error 28: Failed to connect to exampleprovider.com port 443: Connection timed out (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://exampleprovider.com/examplebucket",`
    
The trace points to `/var/www/nextcloud/lib/private/Files/ObjectStore/S3ObjectTrait.php`, where the `verify_bucket_exists` parameter is defined.

In our config.php (which was built after the sample) we have:

```
<?php

$${DUMMY_VAR}CONFIG = [
  'objectstore' => [
    'class' => '\\OC\\Files\\ObjectStore\\S3',
    'arguments' => [
      'bucket' => 'examplebucket',
      'autocreate' => false,
```
In our case the autocreate is ignored regardless of it's state, thus we got attempts of bucket recreation.

**Comment by juliushaertl** [comment]
Fixed by 2281e388b0
